crates/burn-import/src/onnx/op_configuration.rs

```diff
@@ -745,6 +745,64 @@ pub fn layer_norm_config(node: &Node) -> (LayerNormConfig, bool) {
     )
 }
 
+/// Create a PadConfig from the attributes of the node
+pub fn pad_config(node: &Node) -> PadConfig {
+    fn get_pads(node: &Node) -> Vec<usize> {
+        if node.inputs.len() != 3 {
+            panic!("Pad: must provide three inputs")
+        }
+
+        let input_dim = match &node.inputs.first().unwrap().ty {
+            ArgType::Tensor(tensor) => tensor.dim,
+            _ => panic!("Pad: Only tensor input is valid"),
+        };
+
+        let pads: Vec<usize> = match &node.inputs[1].value {
+            Some(Data::Int64s(shape)) => shape
+                .iter()
+                .map(|&x| {
+                    if x < 0 {
+                        // TODO: support negative pads
+                        panic!("Pad: Negative pad is not supported");
+                    }
+                    x as usize
+                })
+                .collect(),
+            _ => panic!("Pad: pads data type must be int64"),
+        };
+
+        if pads.len() != input_dim * 2 {
+            panic!("Pad: pads should be a 1D tensor of shape [2 * num_axes]");
+        }
+        // TODO: Burn's pad should support 1D tensor
+        if input_dim < 2 {
+            panic!("Pad: input tensor should be rank 2 or higher");
+        }
+
+        if input_dim > 2 {
+            log::warn!("Pad: padding will only be applied to the last two dimensions");
+        }
+
+        let left = pads[input_dim - 1];
+        let top = pads[input_dim - 2];
+        let right = pads[pads.len() - 1];
+        let bottom = pads[pads.len() - 2];
+        vec![left, right, top, bottom]
```

Comment:
I think the correct conversion when parsing `pads` should check the pad order `[x1_begin, x2_begin, ..., x1_end, x2_end, ...]`, and we should make sure that all dimensions outside of the last two are not padded otherwise the conversion is not valid (and thus, panic instead of warn).

Following the given [spec example](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Pad), a 4d input with padded on the last two dimensions should have `pads = [0, 0, x2_begin, x3_begin, 0, 0, x2_end, x3_end]` (i.e., `x1_begin`, `x1_end`, `x2_begin`, and `x2_end` are all zero)

Reply:
ok that's a good idea

Reply:
I have pushed the update